distribution/docs/src/main/resources/_contents/_resources/developing-resource-readers-contents.adoc

```diff
@@ -0,0 +1,132 @@
+==== Developing Resource Readers
+
+A `ResourceReader` is a class that retrieves a resource or product from a native/external source and returns it to ${branding}.
+A simple example is that of a File `ResourceReader`.
+It takes a file from the local file system and passes it back to ${branding}.
+New implementations can be created in order to support obtaining Resources from various Resource data stores. 
+
+===== Creating a New `ResourceReader`
+
+Complete the following procedure to create a `ResourceReader`.
+
+. Create a Java class that implements the `${ddf-branding}.catalog.resource.ResourceReader` interface.
+. Deploy the OSGi bundled packaged service to the ${branding} run-time.
+
+===== Implementing the `ResourceReader` Interface
+
+[source,java,linenums]
+----
+public class TestResourceReader implements ${ddf-branding}.catalog.resource.ResourceReader
+----
+
+`ResourceReader` has a couple of key methods where most of the work is performed.
+
+[NOTE]
+====
+*URI* +
+It is recommended to become familiar with the Java API URI class in order to properly build a `ResourceReader`. 
+Furthermore, a URI should be used according to its http://www.w3.org/Addressing/URL/uri-spec.html[specification].
+====
+
+===== retrieveResource
+
+[source,java,linenums]
+----
+public ResourceResponse retrieveResource( URI uri, Map<String, Serializable> arguments )throws IOException, ResourceNotFoundException, ResourceNotSupportedException;
+----
+
+This method is the main entry to the `ResourceReader`.
+It is used to retrieve a `Resource` and send it back to the caller (generally the `CatalogFramework`).
+Information needed to obtain the entry is contained in the `URI` reference.
+The URI Scheme will need to match a scheme specified in the `getSupportedSchemes` method.
+This is how the CatalogFramework determines which `ResourceReader` implementation to use. 
+If there are multiple `ResourceReaders` supporting the same scheme, these `ResourceReaders` will be invoked iteratively. 
+Invocation of the `ResourceReaders` stops once one of them returns a `Resource`.
+
+Arguments are also passed in.
+These can be used by the `ResourceReader` to perform additional operations on the resource.
+
+An example of how `URLResourceReader` (located in the source code at `/trunk/${ddf-branding}/catalog/resource/URLResourceReader.java`) implements the `getResource` method.
```

Comment:
we can remove this example sentence. The location is out of date and the `getResource` method does not exist anymore as it was replaced with `retrieveResource`. We could still mention the URLResourceReader as an example ResourceReader